### PR TITLE
Copter: move brake state into a structure

### PR DESCRIPTION
**Pull request by peterbarker**
This just throws all of the poshold-brake related fields into a structure.

There are methods that could be moved onto that structure (and some of the brake fields then be made private).

But this is just simple.

### CODE REVIEW

**Comment by rmackay9** [comment]
you'll have to test it though!

**Comment by peterbarker** [comment]
I did test it when I originally made the patch a couple of years ago.

I'll retest before the devcall at least.

**Review by khancyr — APPROVED** [review]
Looks good. The testing is already covered by the compiler on this.
otherwise we could just add a simple move in poshold autotest

File: ArduCopter/mode_poshold.cpp [hunk 1]
```diff
@@ -41,8 +41,7 @@ bool ModePosHold::init(bool ignore_checks)
     pilot_roll = 0.0f;
     pilot_pitch = 0.0f;
 
-    // compute brake_gain
```
Comment (rmackay9) [inline ArduCopter/mode_poshold.cpp:44]: It's not clear what's changed here.. but let's not remove the comment.

**Comment by rmackay9** [comment]
Sorry to be grumpy but in general I'm not a huge fan of these types of changes.  It's adding churn and we're all quite busy with more important things.  It's probably OK but now I have to go through this and make sure there are no mistakes.  There are some changes in the scope of variables so it's not just a straight substitution.  Testing has only been done with autotest instead of a detailed review of how it could brake based on these changes.

File: ArduCopter/mode_poshold.cpp (hunk 1 above)
Comment (peterbarker) [reply ArduCopter/mode_poshold.cpp:44]: Comment is back.

**Comment by peterbarker** [comment]
Gave it a quick spin in SITL.  Works just the same in master.

image

(that's master - roll gets perturbed even when only pitch is being played with, which is interesting).

Went through and verified that apart from making one variable a local it's just a name substitution.

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**